File: backend/src/main/kotlin/fr/gouv/cnsp/monitorfish/domain/entities/controls/Control.kt
package fr.gouv.cnsp.monitorfish.domain.entities.controls
import java.time.ZonedDateTime


data class Control(
        var id: Int,
        var vesselId: Int,
        var controller: Controller,
        var controlType: String? = null,
        var controlDatetimeUtc: ZonedDateTime? = null,
        var inputStartDatetimeUtc: ZonedDateTime? = null,
        var inputEndDatetimeUtc: ZonedDateTime? = null,
        var facade: String? = null,
        var longitude: Double? = null,
        var latitude: Double? = null,
        var portLocode: String? = null,
        var portName: String? = null,
        var missionOrder: Boolean? = null,
        var vesselTargeted: Boolean? = null,
        var cnspCalledUnit: Boolean? = null,
        var cooperative: Boolean? = null,
        var preControlComments: String? = null,
        var infraction: Boolean? = null,
        var infractions: List<Infraction> = listOf(),
        var diversion: Boolean? = null,
        var escortToQuay: Boolean? = null,
        var seizure: Boolean? = null,
        var seizureComments: String? = null,
        var postControlComments: String? = null,
        var gearControls: List<GearControl> = listOf())

File: frontend/src/domain/types/control.ts
export type Infraction = {
  infraction: string
  infractionCategory: string
  natinfCode: string
  regulation: string
}

export type GearControl = {
  controlledMesh: number
  declaredMesh: number
  gearCode: string
  gearName: string
  gearWasControlled: boolean
}

export type Controller = {
  administration: string
  controller: string
  controllerType: string
}

export type VesselControl = {
  cnspCalledUnit: boolean
  controlDatetimeUtc: string
  controlType: string
  controller: Controller
  cooperative: boolean
  diversion: boolean
  escortToQuay: boolean
  facade: string
  gearControls: GearControl[]
  infraction: boolean
  infractions: Infraction[]
  inputEndDatetimeUtc: string
  inputStartDatetimeUtc: string
  latitude: string
  longitude: string
  missionOrder: string
  portLocode: string
  portName: string
  postControlComments: string
  preControlComments: string
  seizure: boolean
  seizureComments: string
  vesselTargeted: boolean
}

export type ControlSummary = {
  controls: VesselControl[]
  numberOfAerialControls: number
  numberOfDiversions: number
  numberOfEscortsToQuay: number
  numberOfFishingInfractions: number
  numberOfLandControls: number
  numberOfSeaControls: number
  numberOfSecurityInfractions: number
  numberOfSeizures: number
  vesselId: number
}

// TODO Check that these comments are not some sort of enum value.
export type LastControls = {
  // Contrôle à la débarque
  LAND: ControlAndText
  // Contrôle_en_mer
  SEA: ControlAndText
}

export type ControlAndText = {
  control: VesselControl
  text: string
}
